File: CS/DS12188/MyAppointmentForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraScheduler.UI;
using DevExpress.XtraScheduler;

namespace DS12188 {
    public partial class MyAppointmentForm : AppointmentForm{
        public MyAppointmentForm() {
            InitializeComponent();
        }

        private SchedulerStorage actualStorage;
        private SchedulerStorage filteredStorage;

        private void PrepareFilteredStorage(SchedulerStorage storage) {
            actualStorage = storage;
            filteredStorage = new SchedulerStorage();
            filteredStorage.Appointments.Labels.Clear();
            foreach (AppointmentLabel label in actualStorage.Appointments.Labels) {
                if (label.DisplayName != "Deleted Label")
                    filteredStorage.Appointments.Labels.Add(label);
            }
        }

        public MyAppointmentForm(SchedulerControl control, Appointment apt)
			: base(control, apt, false) {
		}

        public MyAppointmentForm(SchedulerControl control, Appointment apt, bool openRecurrenceForm)
            : base(control, apt, openRecurrenceForm) 
        {            
            
        }

        protected override void UpdateForm() {
            base.UpdateForm();
            PrepareFilteredStorage(Control.Storage);
            AppointmentLabel lbl = actualStorage.Appointments.Labels[Controller.LabelId];
            int index = GetFilteredLabelIndex(lbl);
            edtLabel.Storage = filteredStorage;
            edtLabel.Label = filteredStorage.Appointments.Labels[index];
        }

        protected override void edtLabel_EditValueChanged(object sender, EventArgs e) {
            int index = GetActualLabelIndex(edtLabel.Label);
            Controller.SetLabel(actualStorage.Appointments.Labels[index]);
        }

        private int GetFilteredLabelIndex(AppointmentLabel lbl) {
            return filteredStorage.Appointments.Labels.IndexOf(lbl);
        }

        private int GetActualLabelIndex(AppointmentLabel lbl) {
            return actualStorage.Appointments.Labels.IndexOf(lbl);
        }
    }
}

File: VB/DS12188/MyAppointmentForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraScheduler.UI
Imports DevExpress.XtraScheduler

Namespace DS12188
	Partial Public Class MyAppointmentForm
		Inherits AppointmentForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private actualStorage As SchedulerStorage
		Private filteredStorage As SchedulerStorage

		Private Sub PrepareFilteredStorage(ByVal storage As SchedulerStorage)
			actualStorage = storage
			filteredStorage = New SchedulerStorage()
			filteredStorage.Appointments.Labels.Clear()
			For Each label As AppointmentLabel In actualStorage.Appointments.Labels
				If label.DisplayName <> "Deleted Label" Then
					filteredStorage.Appointments.Labels.Add(label)
				End If
			Next label
		End Sub

		Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment)
			MyBase.New(control, apt, False)
		End Sub

		Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment, ByVal openRecurrenceForm As Boolean)
			MyBase.New(control, apt, openRecurrenceForm)

		End Sub

		Protected Overrides Sub UpdateForm()
			MyBase.UpdateForm()
			PrepareFilteredStorage(Control.Storage)
			Dim lbl As AppointmentLabel = actualStorage.Appointments.Labels(Controller.LabelId)
			Dim index As Integer = GetFilteredLabelIndex(lbl)
			edtLabel.Storage = filteredStorage
			edtLabel.Label = filteredStorage.Appointments.Labels(index)
		End Sub

		Protected Overrides Sub edtLabel_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
			Dim index As Integer = GetActualLabelIndex(edtLabel.Label)
			Controller.SetLabel(actualStorage.Appointments.Labels(index))
		End Sub

		Private Function GetFilteredLabelIndex(ByVal lbl As AppointmentLabel) As Integer
			Return filteredStorage.Appointments.Labels.IndexOf(lbl)
		End Function

		Private Function GetActualLabelIndex(ByVal lbl As AppointmentLabel) As Integer
			Return actualStorage.Appointments.Labels.IndexOf(lbl)
		End Function
	End Class
End Namespace